examples/api/lib/cupertino/sheet/cupertino_sheet.3.dart

```diff
@@ -0,0 +1,114 @@
+}
+
+class CupertinoSheetApp extends StatelessWidget {
+  const CupertinoSheetApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return const CupertinoApp(title: 'Scrollable Cupertino Sheet', home: HomePage());
+  }
+}
+
+class HomePage extends StatelessWidget {
+  const HomePage({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return CupertinoPageScaffold(
+      navigationBar: const CupertinoNavigationBar(
+        middle: Text('Scrollable Cupertino Sheet Example'),
+        automaticBackgroundVisibility: false,
+      ),
+      child: Center(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            CupertinoButton.filled(
+              onPressed: () {
+                Navigator.of(context).push(
+                  CupertinoSheetRoute<void>.scrollable(
+                    scrollableBuilder: (BuildContext context, ScrollController controller) =>
+                        _ScrollableSheetBody(scrollController: controller),
+                  ),
+                );
+              },
+              child: const Text('Open Sheet'),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class _ScrollableSheetBody extends StatelessWidget {
+  const _ScrollableSheetBody({required this.scrollController});
+
+  final ScrollController scrollController;
+
+  @override
+  Widget build(BuildContext context) {
+    return CupertinoPageScaffold(
+      navigationBar: CupertinoSheetNavbar(
+        child: CupertinoNavigationBar(
+          backgroundColor: CupertinoColors.systemGrey3,
+          middle: const Text('Scrollable Sheet'),
+          automaticBackgroundVisibility: false,
+          leading: CupertinoButton(
+            padding: EdgeInsets.zero,
+            child: const Text('Close'),
+            onPressed: () {
+              CupertinoSheetRoute.popSheet(context);
+            },
+          ),
+        ),
+      ),
+      child: CustomScrollView(
+        controller: scrollController,
+        primary: false,
+        slivers: <Widget>[
+          SliverList(
+            delegate: SliverChildBuilderDelegate((BuildContext context, int index) {
+              return Container(
+                alignment: Alignment.center,
+                height: 100,
+                child: const Text('Scroll Me'),
+              );
+            }, childCount: 20),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class CupertinoSheetNavbar extends StatelessWidget implements ObstructingPreferredSizeWidget {
+  const CupertinoSheetNavbar({super.key, required this.child});
+
+  final CupertinoNavigationBar child;
+
+  @override
+  bool shouldFullyObstruct(BuildContext context) {
+    return child.shouldFullyObstruct(context);
+  }
+
+  @override
+  Size get preferredSize {
+    return child.preferredSize;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return CupertinoSheetDragArea(child: child);
```

Comment:
Could this be automatically built into CupertinoNavigationBar?

Reply:
Would it be easier for the developer if, in the case of a scrollable sheet, we provide a header parameter and we can manage the gesture resolution ourselves internally?
What is the native experience like for developers building the UI you described above?

Reply:
> I believe in that case then a scroll would be triggered on a drag upwards originating in the pinned sliver, instead of the stretch animation.

Did you try it?

We have control over how the scrolling offset is applied here with the custom scroll position class in the scrollable sheet, could we divert it in this case?